client/clientapi.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/codegangsta/cli"
+)
+
+type Plugin struct {
+	Name        string `json:"name"`
+	FullName    string `json:"full_name"`
+	Type        string `json:"type"`
+	Owner       string `json:"owner"`
+	Description string `json:"description"`
+	URL         string `json:"url"`
+	Forks       int    `json:"fork_count"`
+	Stars       int    `json:"star_count"`
+	Watchers    int    `json:"watch_count"`
+	Issues      int    `json:"issues_count"`
+}
+
+func printPlugin(p Plugin) {
+	fmt.Printf("Name: %s\n", p.Name)
+	fmt.Printf("FullName: %s\n", p.FullName)
+	fmt.Printf("Type: %s\n", p.Type)
+	fmt.Printf("Owner: %s\n", p.Owner)
+	fmt.Printf("Description: %s\n", p.Description)
+	fmt.Printf("URL: %s\n", p.URL)
+	fmt.Printf("Forks: %d\n", p.Forks)
+	fmt.Printf("Stars: %d\n", p.Stars)
+	fmt.Printf("Watchers: %d\n", p.Watchers)
+	fmt.Printf("Issues: %d\n\n", p.Issues)
+}
+
+func printCollectors(plugins []Plugin) {
+	fmt.Println("\n========Collectors========")
+	for _, v := range plugins {
+		if v.Type == "collector" {
+			fmt.Println(v.Name)
+		}
+	}
+}
+
+func printProcessors(plugins []Plugin) {
+	fmt.Println("\n========Processors========")
+	for _, v := range plugins {
+		if v.Type == "processor" {
+			fmt.Println(v.Name)
+		}
+	}
+}
+
+func printPublishers(plugins []Plugin) {
+	fmt.Println("\n========Publishers========")
+	for _, v := range plugins {
+		if v.Type == "publisher" {
+			fmt.Println(v.Name)
+		}
+	}
+}
+
+func infoByName(ctx *cli.Context) {
+	if len(ctx.Args()) > 1 {
+		fmt.Println("Incorrect usage of list command--please enter in format: plugin list [plugin_name]")
+	} else {
+		resp, err := http.Get("http://localhost:8080/plugins/")
+		if err != nil {
+			fmt.Printf("error is: %s\n", err)
+		} else {
+			body, err := ioutil.ReadAll(resp.Body)
+			plugins := make([]Plugin, 0)
```

Comment:
add an error handling after reading the body
```
if err != nil {
 				fmt.Println(err)
 			}
```

NB:
You can move the `defer resp.Body.Close()` just after handling the error.